includes/admin/extensions/abstract-extension.php

```diff
@@ -0,0 +1,284 @@
+<?php
+
+namespace EDD\Admin\Extensions;
+
+use \EDD\Admin\Pass_Manager;
+
+abstract class Extension {
+
+	/**
+	 * The product ID. This only needs to be set if the extending class is
+	 * for a single product.
+	 *
+	 * @since 2.11.x
+	 * @var int
+	 */
+	protected $item_id;
+
+	/**
+	 * The required AA pass level.
+	 */
+	const PASS_LEVEL = Pass_Manager::PERSONAL_PASS_ID;
+
+	/**
+	 * The Extension Manager
+	 *
+	 * @var \EDD\Admin\Extensions\Extension_Manager
+	 */
+	protected $manager;
+
+	public function __construct() {
+		$this->manager = new \EDD\Admin\Extensions\Extension_Manager( static::PASS_LEVEL );
+	}
+
+	/**
+	 * Whether the extension is activated.
+	 *
+	 * @since 2.11.x
+	 * @return bool
+	 */
+	abstract protected function is_activated();
+
+	/**
+	 * Output the settings field (installation helper).
+	 *
+	 * @param array $args
+	 * @return void
+	 */
+	public function settings_field() {
+		if ( $this->is_activated() ) {
+			return;
+		}
+		$this->do_single_extension_card();
+	}
+
+	/**
+	 * Outputs a single extension card.
+	 *
+	 * @since 2.11.x
+	 * @param bool|int $item_id Optional: the individual extension product ID.
```

Comment:
```suggestion
	 * @param false|int $item_id Optional: the individual extension product ID.
```

I think this is either `false` or an item ID, but never `true`? If so, I think saying `false` explicitly is a bit clearer.